### PR TITLE
Split contribution instructions out to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by fitzgen**
Also update the set up and testing instructions.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Contributing to `wasmtime-py`
+
+`wasmtime-py` is a [Bytecode Alliance] project. It follows the Bytecode
+Alliance's [Code of Conduct] and [Organizational Code of Conduct].
+
+So far this extension has been written by folks who are primarily Rust
+programmers, so it's highly likely that there's some faux pas in terms of Python
+idioms. Feel free to create a PR to help make things more idiomatic if you see
+something!
```
Comment (whitequark) [inline CONTRIBUTING.md:9]: I'm pretty sure that by this point, this extension does qualify as "idiomatic Python"!

Comment (fitzgen) [reply CONTRIBUTING.md:9]: Heh yeah I'm happy to remove this clause if its unnecessary now (I'm also not really in a position to judge whether this package is idiomatic or not :sweat_smile: )

Comment (whitequark) [reply CONTRIBUTING.md:9]: I'm sure there are a few things that could be further improved, but Python packages aren't all that consistent in general. Wasmtime-py seems at the very least better than average to me, which means the warning no longer applies!

**Comment by alexcrichton** [comment]
I've pushep up a few wording tweaks which makes the ask for help on idioms a bit less zealous (but still there because I'm paranoid), and I've also freshened up the instructions for handling the native lib.

**Pull request merged**